File: fine_tuning/results/test_output_files/question-73-pass-1.py
from math import sqrt


def isoctagonal(N):
    n = int(sqrt(N))
    if (n * (n + 1) / 2 == N):
        print("Yes")
    else:
        print("No")


if __name__ == "__main__":
    N = 12
    isoctagonal(N)

File: results/feedback_pipeline/cpp_results/cpp_files_after_feedback/question-73-pass-1.cpp
#include <bits/stdc++.h>
using namespace std;
bool isOctagonal(int N) {
    int n = (int)sqrt(24*N+1);
    return (n*(n-1)) == 8*N;
}
int main() {
    int N;
    cin >> N;
    if (isOctagonal(N)) {
        cout << "Yes, " << N << " is an octagonal number." << endl;
    } else {
        cout << "No, " << N << " is not an octagonal number." << endl;
    }
    return 0;
}